components/ShadowLayer/examples/ShadowAnimationExampleViewController.m

```diff
@@ -0,0 +1,259 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "ShadowAnimationExampleViewController.h"
+
+#import "MaterialAnimationTiming.h"
+#import "MaterialButtons+ButtonThemer.h"
+#import "MaterialButtons.h"
+#import "MaterialShadowLayer.h"
+
+static const CGFloat kAnimationDuration = (CGFloat)2.0;
+static const CGFloat kStartCornerRadius = (CGFloat)0.001;
+static const CGFloat kEndCornerRadius = (CGFloat)20.0;
+
+@interface MDCShadowLayer (CustomAnimation)
+
+@property(nonatomic, strong) CAShapeLayer *topShadow;
+@property(nonatomic, strong) CAShapeLayer *bottomShadow;
+@property(nonatomic, strong) CAShapeLayer *topShadowMask;
+@property(nonatomic, strong) CAShapeLayer *bottomShadowMask;
```

Comment:
We generally discourage relying on private API implementation details for our components.

If access to these layers is a requirement for implementing this animation, then it may actually be more sustainable long-term to implement the solution as an animation API on the shadow layer class. Consider, for example, that in the future we might move to a simpler shadow implementation that does not rely on a top and bottom shadow and how that would affect this example as-is.